### ISSUE TITLE
Add table support to `ConvertFrom-String`

### ISSUE THREAD

**Issue by iRon7** [PowerShell/PowerShell#16677]
### Summary of the new feature / enhancement

This issue is related to #16298 [`String` >>>Regex>>> `PSCustomObject`](https://github.com/PowerShell/PowerShell/issues/16257) which is closed after it moved to a discussion (not sure about the exact reason it is closed, meaning that I might make the same mistake here).

Anyways, reading from a fix text table usually generated from an external program is something that is quiet often required noting the concerned StackOverflow questions, like:
* [Convert Formatted String Output with empty cells to PS Object](https://stackoverflow.com/a/69118599/1701026)
* [Powershell String to csv (netstat -ano)](https://stackoverflow.com/q/65346664/1701026)
* [convert list output to table(object)](https://stackoverflow.com/a/60882963/1701026)
* [Convert System.String to object](https://stackoverflow.com/q/56964468/1701026)
* [Another grep / awk q, parsing diskpart output](https://stackoverflow.com/q/42680382/1701026)
* [How to read powershell table from file](https://stackoverflow.com/a/53364959/1701026)
* [Split string on arbitrary-length substrings (Powershell)](https://stackoverflow.com/a/55752375/1701026)
* [ConvertFrom-Sting Template Parsing Failure with Array](https://stackoverflow.com/q/52835623/1701026)
* [How can I use powershell to intercept a specified string of a command output?](https://stackoverflow.com/a/51279071/1701026)

As you can see from some of the answers, I have created my own "solution" for this: [`ConvertFrom-SourceTable`](https://www.powershellgallery.com/packages?q=convertfrom-sourcetable) but I would expect a more standard way using the `ConvertFrom-String` with e.g. a `-Table` parameter.

Besides from converting the output of an external program, it could be useful to build a human readable" in-script source/settings table.

### Proposed technical implementation details (optional)

```PowerShell
$Colors = ConvertFrom-SourceTable ' # ConvertFrom-String -Table
	Name       Value         RGB
	----       -----         ---
	Black   0x000000       0,0,0
	White   0xFFFFFF 255,255,255
	Red     0xFF0000     255,0,0
	Lime    0x00FF00     0,255,0
	Blue    0x0000FF     0,0,255
	Yellow  0xFFFF00   255,255,0
	Cyan    0x00FFFF   0,255,255
	Magenta 0xFF00FF   255,0,255
	Silver  0xC0C0C0 192,192,192
	Gray    0x808080 128,128,128
	Maroon  0x800000     128,0,0
	Olive   0x808000   128,128,0
	Green   0x008000     0,128,0
	Purple  0x800080   128,0,128
	Teal    0x008080   0,128,128
	Navy    0x000080     0,0,128'
```

**Comment by iSazonov** [comment]
@iRon7 The cmdlet is not in the repository and we can do nothing here. We already have #726 as tracking issue and it is better to have all other new conversations in Discussions with reference to #726.

**Comment by iRon7** [comment]
@iSazonov, Thank you for the explanation.
I have closed the issue as is linked to #726 and the idea is recorded anyways in case anything changes to the concerned `ConvertFrom-String` cmdlet.
